src/kemal/handler.cr

```diff
@@ -15,6 +15,7 @@ class Kemal::Handler < HTTP::Handler
 
   def add_route(method, path, &handler : Kemal::Context -> _)
     @routes << Route.new(method, path, &handler)
+    @routes << Route.new("HEAD", path, &handler) if method == "GET"
```

Comment:
That's great. Basically, I would like to change route matching implementation a bit. Instead of matching coming URL with defined routes, I would to implement it in such a way that for obvious routes such as `/login`, `/signup`, route get matched in `O(1)`. And for remaining routes we can use loop thing.